docs/how-to-guides/integrating-autoware/creating-vehicle-interface-package/creating-vehicle-interface-for-ackerman-kinematic-model.md

```diff
@@ -1,5 +1,198 @@
-# Creating vehicle interface for ackerman kinematic model
+# Creating vehicle interface for Ackermann kinematic model
 
-!!! warning
+This page introduces a module vehicle_interface and explains how to implement it.
 
-    Under Construction
+## What is a vehicle interface
+
+`vehicle interface` is an interface that connects the control commands and your hardware.  
+Autoware publishes control commands such as
+
+- Longitudinal control
+- Steering control
+- Car light commands
+
+Then, `vehicle interface` converts these commands into actuation such like
+
+- Motors and breaks
+- The steering wheel
+- Lighting control
+
+So think of the vehicle interface as a module that runs the hardware to realize the input commands provided by Autoware.
+
+<figure markdown>
+  ![vehicle_interface_IO](images/Vehicle-Interface-Bus-ODD-Architecture.drawio.svg){ align=center }
+  <figcaption>
+    An example of inputs and outputs for vehicle interface
+  </figcaption>
+</figure>
+
+This page shows you a brief explanation how to implement your vehicle interface, but [you can see further information of vehicle interface in the "design" page](https://autowarefoundation.github.io/autoware-documentation/main/design/autoware-interfaces/components/vehicle-interface/). **Note that there is no package named "vehicle interface" prepared in Autoware. It is a necessary package to actuate your vehicle, but you have to create one by yourself since it is very specific to your hardware.** For example, if you are using a by-wire kit [PACMod](https://autonomoustuff.com/platform/pacmod), a vehicle interface named [`pacmod_interface` published by TIER IV, Inc.](https://github.com/tier4/pacmod_interface/tree/main) is available. However, if you have constructed something original and haven't found an open source vehicle interface applicable, you have to implement your own vehicle interface from scratch.
+
+---
+
+## How to implement a vehicle interface
+
+The following instructions describe how to create a vehicle interface.
+
+<!-- pmarkdownlint-disable MD029 -->
```

Comment:
- typo?
- Do you still need to ignore this rule after applying pre-commit?

Reply:
I'm sorry to keep that left.
This part is already solved after I fixed this document by using subsubsections.
I will delete this comment in the next commit.